ql/src/java/org/apache/hadoop/hive/ql/exec/DDLPlanUtils.java

```diff
@@ -0,0 +1,1005 @@
+
+  private final String EXIST_BIT_VECTORS = "-- BIT VECTORS PRESENT FOR <" + DATABASE_NAME + ">.<" + TABLE_NAME + "> " +
+      "FOR COLUMN <" + COLUMN_NAME + "> BUT THEY ARE NOT SUPPORTED YET. THE BASE64 VALUE FOR THE BITVECTOR IS <" +
+      BASE_64_VALUE +"> ";
+
+  private final String EXIST_BIT_VECTORS_PARTITIONED = "-- BIT VECTORS PRESENT FOR <" + DATABASE_NAME + ">.<" +
+      TABLE_NAME + "> PARTITION <" + PARTITION_NAME + "> FOR COLUMN <"
+      + COLUMN_NAME + "> BUT THEY ARE NOT SUPPORTED YET.THE BASE64 VALUE FOR THE BITVECTOR IS <" +
+      BASE_64_VALUE +"> ";
+
+  /**
+   * Returns the create database query for a give database name.
+   *
+   * @param dbNames
+   * @return
+   */
+  public List<String> getCreateDatabaseStmt(Set<String> dbNames) {
+    List<String> allDbStmt = new ArrayList<String>();
+    for (String dbName : dbNames) {
+      if (dbName.equals("default")) {
+        continue;
+      }
+      ST command = new ST(CREATE_DATABASE_STMT);
+      command.add(DATABASE_NAME, dbName);
+      allDbStmt.add(command.render());
+    }
+    return allDbStmt;
+  }
+
+  public Map<String, String> getTableColumnsToType(Table tbl) {
+    List<FieldSchema> fieldSchemas = tbl.getAllCols();
+    Map<String, String> ret = new HashMap<String, String>();
+    fieldSchemas.stream().forEach(f -> ret.put(f.getName(), f.getType()));
+    return ret;
+  }
+
+  public List<String> getTableColumnNames(Table tbl) {
+    List<FieldSchema> fieldSchemas = tbl.getAllCols();
+    List<String> ret = new ArrayList<String>();
+    fieldSchemas.stream().forEach(f -> ret.add(f.getName()));
+    return ret;
+  }
+
+  public String getPartitionActualName(Partition pt) {
+    Map<String, String> colTypeMap = getTableColumnsToType(pt.getTable());
+    String[] partColsDef = pt.getName().split(",");
+    List<String> ptParam = new ArrayList<>();
+    for (String partCol : partColsDef) {
+      String[] colValue = partCol.split("=");
+      if (colTypeMap.get(colValue[0]).equalsIgnoreCase("string")) {
+        ptParam.add(colValue[0] + "='" + colValue[1] + "'");
+      } else {
+        ptParam.add(colValue[0] + "=" + colValue[1]);
+      }
+    }
+    return StringUtils.join(ptParam, ",");
+  }
+
+  public boolean checkIfDefaultPartition(String pt){
+    if(pt.contains(HIVE_DEFAULT_PARTITION)){
+      return true;
+    }
+    else {
+      return false;
+    }
+  }
+
+  /**
+   * Creates the alter table command to add a partition to the given table.
+   *
+   * @param pt
+   * @return
+   * @throws MetaException
+   */
+  public String getAlterTableAddPartition(Partition pt) throws MetaException {
+    Table tb = pt.getTable();
+    ST command = new ST(ALTER_TABLE_CREATE_PARTITION);
+    command.add(DATABASE_NAME, tb.getDbName());
+    command.add(TABLE_NAME, tb.getTableName());
+    command.add(PARTITION, getPartitionActualName(pt));
+    if(checkIfDefaultPartition(pt.getName())){
```

Comment:
Can you add a TODO here?
//TODO: adding/updating stats to default partition is not allowed now. Need to fix this later.